src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -173,7 +245,26 @@ internal set
         /// <summary>
         /// Gets or sets the telemetry channel.
         /// </summary>
-        public ITelemetryChannel TelemetryChannel { get; set; }
+        public ITelemetryChannel TelemetryChannel
+        {
+            get
+            {
+                return this.telemetryChannel;
+            }
+
+            set
+            {
+                ITelemetryChannel oldChannel = this.telemetryChannel;
+                this.telemetryChannel = value;
+
+                // If we have a previously assigned channel which is not the same one as the "new" value
+                // passed in then we need to dispose of the old channel to keep from leaking resources.
+                if (oldChannel != null && oldChannel != value)
+                {
+                    oldChannel.Dispose();
```

Comment:
I would be happy to add an optional boolean parameter, defaulted to false, the way StreamReader provides "leaveOpen" for the Stream it wraps in newer versions of .NET.
The current behavior was already to dispose of the channel whether it was passed in or not, and this is consistent with how StreamReader closes (which disposes) the Stream it uses.
So I don't believe we should change that behavior at this moment in time.